Library/Homebrew/system_command.rb

```diff
@@ -76,13 +82,17 @@ def command
   def env_args
     return [] if env.empty?
 
-    variables = env.map do |name, value|
-      sanitized_name = Shellwords.escape(name)
-      sanitized_value = Shellwords.escape(value)
-      "#{sanitized_name}=#{sanitized_value}"
-    end
+    unset_variables = env.select { |_, value| value.nil? }
+                         .flat_map { |name,| ["-u", name] }
+
+    set_variables = env.reject { |_, value| value.nil? }
+                       .flat_map do |name, value|
+                         sanitized_name = Shellwords.escape(name)
```

Comment:
Can you indent this two be further left? It's not clear it's a multiline block as-is